search-app/src/cmr/search/api/routes.clj

```diff
@@ -315,20 +305,46 @@
      :headers {common-routes/CONTENT_TYPE_HEADER (mt/with-utf-8 mt/json)}
      :body results}))
 
-(def robots-txt-response
-  "Returns the robots.txt response."
-  {:status 200
-   :body (slurp (io/resource "public/robots.txt"))})
+(defn find-query-str-mixed-arity-param
+  "Return the first parameter that has mixed arity, i.e., appears with both single and multivalued in
+  the query string. e.g. foo=1&foo[bar]=2 is mixed arity, so is foo[]=1&foo[bar]=2. foo=1&foo[]=2 is
+  not. Parameter with mixed arity will be flagged as invalid later."
+  [query-str]
+  (when query-str
+    (let [query-str (-> query-str
+                        (str/replace #"%5B" "[")
+                        (str/replace #"%5D" "]")
+                        (str/replace #"\[\]" ""))]
+      (last (some #(re-find % query-str)
+                  [#"(^|&)(.*?)=.*?\2\["
```

Comment:
Seriously, though, the rule we go by is that you can't (knowingly) merge broken code, even if you didn't write it originally. It's a drag, but it keeps our code cleaner. (Leo has had to fix my mistakes countless times).

I think you can just put an ampersand before the back references like this
`#"(^|&)(.*?)=.*?&\2\["`
and this
`#"(^|&)(.*?)\[.*?&\2="`
and it will fix the issue.